lib/scanner/index.js

```diff
@@ -156,7 +157,18 @@ class Scanner {
         this.callback(data);
       })
 
-      .catch(err => console.error(err));
+      .catch(err => {
+        console.info(err);
+        if (this.netErrorListener) {
+          this.netErrorListener(err);
+        }
+      });
+  }
+
+  set networkErrorListener(cb) {
```

Comment:
Ah I get it now. See [comment](https://github.com/mozilla-magnet/magnet/pull/39/files#r60883335)